.travis.yml

```diff
@@ -0,0 +1,42 @@
+sudo: false
+
+addons:
+  apt:
+    packages:
+      - oracle-java8-installer
+
+language: java
+
+dist: trusty
+
+jdk:
+  - openjdk8
+
+install: true # skips travis' default installation step which executes gradle assemble.
+
+script: ./gradlew clean build
+
+# The before_cache and the cache steps cache the gradle installation on travis.
+before_cache:
```

Comment:
ok, then 
```
cache:
 +  directories:
 +    - $HOME/.gradle/caches/
 +    - $HOME/.gradle/wrapper/
```

should be on top. but anyway i'd keep them both on top, since it is just a setup, keeping the real build logic on bottom.